src/main/java/ca/uhn/fhir/jpa/starter/service/HelperService.java

```diff
@@ -1627,6 +1709,184 @@ private <R extends IBaseResource> String updateResource(String keycloakId, IBase
 		return null;
 	}
 
+	private <R extends IBaseResource> String createResource(String keycloakId, IBaseResource resource, Class<R> resourceClass) {
+
+		RealmResource realmResource = FhirClientAuthenticatorService.getKeycloak().realm(appProperties.getKeycloak_Client_Realm());
+		R existingResource = null;
+		GroupResource groupResource = null;
+		GroupRepresentation group = null;
+		try {
+			if (resourceClass.equals(Organization.class)){
+				groupResource = realmResource.groups().group(keycloakId);
+				group = groupResource.toRepresentation();
+				String orgId = group.getAttributes().get("organization_id").get(0);
+				existingResource = FhirClientAuthenticatorService.getFhirClient().read().resource(resourceClass).withId(orgId).execute();
+			} else if (resourceClass.equals(Location.class)) {
+				groupResource = realmResource.groups().group(keycloakId);
+				group = groupResource.toRepresentation();
+				String locId = group.getAttributes().get("location_id").get(0);
+				existingResource = FhirClientAuthenticatorService.getFhirClient().read().resource(resourceClass).withId(locId).execute();
+			}
+		} catch (ResourceNotFoundException e) {
+			logger.warn("RESOURCE NOT FOUND");
+		}
+		try {
+			if (existingResource == null) {
+			Method addIdentifier = resource.getClass().getMethod("addIdentifier");
+			Identifier obj = (Identifier) addIdentifier.invoke(resource);
+			obj.setSystem(IDENTIFIER_SYSTEM + "/KeycloakId");
+			obj.setValue(keycloakId);
+			MethodOutcome outcome = FhirClientAuthenticatorService.getFhirClient().update().resource(resource).execute();
+			return outcome.getId().getIdPart();
```

Comment:
Is there a possibility for the outcome to be and error instead of a resource? eg: When we PUT objects the outcome maybe an OperationOutcome of some sort that is an error instead of the resource that has been updated on the FHIR server